core/pkg/artifacts/linker.go

```diff
@@ -72,3 +81,50 @@ func (al *ArtifactLinker) Link() error {
 
 	return err
 }
+
+// resolveOrgEntityName take an Entity and an potentially empty Organization(display org name or org entity name)
+// and using those inputs we fetch the org entity and:
+// 1. validate the user inputted the correct org name or org entity name for the link path
+// 2. return the org entity name in place of if no org was inputted or the display org name
+// so we can use the correct entity name to link the artifact.
```

Comment:
almost there; doc comments should be very tidy because we spend a lot of time reading them. Conciseness and clarity are important as are spelling and formatting.

```suggestion
// resolveOrgEntityName fetches the portfolio's org entity's name.
//
// The organization parameter may be empty, an org's display name, or an org entity name.
//
// If the server doesn't support fetching the org name of a portfolio, then this returns
// the organization parameter, or an error if it is empty. Otherwise, this returns the
// fetched value after validating that the given organization, if not empty, matches
// either the org's display or entity name.
```